### PR TITLE
[UI Tests] Make current WordPress UI Tests to run against Jetpack app locally

### PR DESCRIPTION
**Pull request by tiagomar**
### What
The objective of this PR is to make the current Wordpress UI Tests to run against the Jetpack app locally. 

### How
A `JetpackUITests` test plan was created similar to the existing `WordPressUITests`, but targeting the Jetpack app (Kudos to @jostnes for further investigating it when I thought it was not possible to leverage the same files, even though it didn't make sense to me)

Some tests and a XCTest+Extensions needed adjustments to either be able to run for both apps or to run on JP only, since features are about to be removed from WP app.

[MainNavigationTests.swift]([WordPress/WordPressUITests/Tests/MainNavigationTests.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/files?file-filters%5B%5D=.swift&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xctestplan&show-viewed-files=true#diff-4bc751f92978a61af7d3d2a82da30de4fe53def339f8d97e0af0b98182ba3e03)) - Small change due to the fact that the JP opens in the `Home` tab by default while WP opens the `Menu`.

[StatsTests.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/files?file-filters%5B%5D=.swift&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xctestplan&show-viewed-files=true#diff-aa1f96f94d1ff8535f93494403c70d30aaa943211f79830c1a629949b1d864cc) - JP has a revamped Stats that didn't go to WP. With the revamp, the `Insights` tab changed and now displays different data, which was reflected in the test's expected data. Since `Stats` is being removed from WP next week, I thought it didn't worth the effort making the test run for both, so I just disabled `Stats` tests for WP.

[SupportScreenTests.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/files?file-filters%5B%5D=.swift&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xctestplan&show-viewed-files=true#diff-d55595f0cfac7390cdc253d501abb8ac5db09159775cdc9563b2f2cc7387002b) | [SupportScreen.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/files?file-filters%5B%5D=.swift&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xctestplan&show-viewed-files=true#diff-4e3360cb13875012daef30636d83aac2cb35e66b8dc469e46f7a9425e8c3ac5d) - While it looks like a new test being added, it's just bringing back a previously existing test.  `Contact Us` screen has recently moved from WP to JP and at that moment `testContactUsCanBeLoadedDuringLogin()` was converted into `testSupportForumsCanBeLoadedDuringLogin()`. Now `testContactUsCanBeLoadedDuringLogin()` is back for JP and disabled for WP, the same way as `testSupportForumsCanBeLoadedDuringLogin()` is enabled only for WP. 

[XCTest+Extensions.swift](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/files?file-filters%5B%5D=.swift&file-filters%5B%5D=.xcscheme&file-filters%5B%5D=.xctestplan&show-viewed-files=true#diff-0e744622889925d683626e21ce5271000e5853df2158900d5eae18df3e9fb15f) - Small change to make the `removeApp` work for both apps.

### Testing
* `WordPressUITests` must pass in CI.
* Run `JetpackUITests` test plan locally and all tests should pass.

### CODE REVIEW

File: WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift
Comment (jostnes) [inline WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: nitpick: we have [the WPUITestCredentials file](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/WordPressUITests/WPUITestCredentials.swift) that store credential strings, i think it might good to keep this in that file or reuse the value already there?

File: WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift
Comment (jostnes) [inline WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: do we know the condition/cause in which contact information is needed? would it be possible to use that condition instead of `...IfNeeded`? 

i think if we know the condition, we have the control to call it only on tests that need it. on tests that don't, we can exclude it and not wait for the condition to happen. wdyt?

**Review by jostnes — APPROVED** [review]
thanks for making this change @tiagomar! i left some comments that are non-blocking and overall i think it's 👍 

approving this, but i think it would be good if we could have another set of eyes review this too before merging (especially around the new scheme and test plan)

File: WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift
Comment (tiagomar) [reply WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: Sure! It makes total sense. Done it in [423b99e](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/commits/423b99e1c18e56ef8e6f95d91da349bc2c5b985f).

File: WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift
Comment (tiagomar) [reply WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: In order to contact the support the user must add contact information. When reaching out for the first time, a pop-up is displayed after tapping the `Contact Support` button asking to enter (for logged out users) or confirm (for logged in users). The next time the user tries to contact support the application uses the stored information, so no pop-up is displayed. So the only way to know if addin the contact information is needed is checking if the pop-up is displayed. The  `...IfNeeded` comes to give some flexibility so whoever is writing a test case doesn't need to care about it.

> i think if we know the condition, we have the control to call it only on tests that need it. on tests that don't, we can exclude it and not wait for the condition to happen. wdyt?

Right now we only have one test covering the Contact Support and we will always need to inform the contact information because 1) the current test covers the user logged out case and 2) since we've been removing and installing the app before each test, even if the test covered a logged in case, it would still see the pop-up. However, if we need to write a new tests that visit the Support screen twice, the second one wouldn’t see the pop-up.

I'm not sure if we would gain much? Let me know if I got you wrong... =]

Thank you! 🙇‍♂️

Comment (jostnes) [reply WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: thanks for the context and explanation @tiagomar!

while it gives the flexibility of not needing to know the condition, i'm always wary of using something like this as it might bloat the test suite with `waits` in the long run. when we end up with a lot of `...IfNeeded` methods for things we know/can control, for cases where it's not needed, there will be additional waiting time that could have been avoided.

in this case, since there's only 1 test case (for now) and the pop-up seems like it's going to be most likely always needed, what do you think about simplifying it to something like (and dropping `addContactInformationIfNeeded` method):
```
    public func contactSupport(userEmail: String) throws -> ContactUsScreen {
        let emailTextField = app.textFields["Email"]

        app.cells["contact-support-button"].tap()
        // email and username pop up displayed on first visit
        emailTextField.tap()
        emailTextField.typeText(userEmail)
        app.buttons["OK"].tap()

        return try ContactUsScreen()
    }

```

i don't think there are any critical scenarios we would want to cover in UI tests where we would need to visit the support screen twice. i could be wrong though! feel free to keep it as is if there are cases that i'm missing/didn't think about.

Comment (tiagomar) [reply WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift]: > i don't think there are any critical scenarios we would want to cover in UI tests where we would need to visit the support screen twice. i could be wrong though! feel free to keep it as is if there are cases that i'm missing/didn't think about.

You're right here. Given that the pop-up is not network triggered, I don't think removing this wait will bring flakiness in, so let's try it out. Done in [`3483025`](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/commits/34830251e83292b43af5981aff22529226241a3e).

File: WordPress/WordPressUITests/Utils/XCTest+Extensions.swift
Comment (mokagio) [inline WordPress/WordPressUITests/Utils/XCTest+Extensions.swift:79]: TIL about `XCUIApplication` `label`. I couldn't find it in the [`XCUIApplication` docs](https://developer.apple.com/documentation/xctest/xcuiapplication) or in its superclass [`XCUIElement`](https://developer.apple.com/documentation/xctest/xcuielement).

Inspecting the source showed it's actually part of [`XCUIElementAttributes`](https://developer.apple.com/documentation/xctest/xcuielementattributes).

**Pull request merged**
